### PR TITLE
refactor: clean up button-group styles

### PR DESCRIPTION
**Pull request by lex111**

### CODE REVIEW

File: packages/core/styles/components/button-group.pcss [hunk 1]
```diff
@@ -6,27 +6,23 @@
...
-      margin-left: var(--ifm-button-group-margin);
     }
 
     &:not(:last-child) {
       border-bottom-right-radius: 0;
       border-top-right-radius: 0;
     }
-
-    &--active {
-      z-index: 1; /* Make the active button's border take precedence. */
```
Comment (lex111) [inline packages/core/styles/components/button-group.pcss:28]: Doesn't make sense, don't know why it's needed at all.

**Review by Josh-Cena — COMMENTED** [review]
Support for `gap` seems inadequate: only 88%. Should we have some policies about what percentage we aim to support? (I also don't know if it can be mitigated through autoprefixing/etc. but from the dist CSS it seems unlikely)

**Comment by lex111** [comment]
It's a pretty big support (eg `margin-top` has 94.5% support, not 100%), all major browsers support it, we already actively use this property in other places, so shouldn't be any problem here.

**Comment by Josh-Cena** [comment]
OK let's do this 👍

`margin-top` not having 100% is because there's unknown support, though, not because it's actually unsupported

**Pull request merged**